lld/MachO/Driver.cpp

```diff
@@ -95,11 +95,16 @@ static std::optional<StringRef> findLibrary(StringRef name) {
               findPathCombination("lib" + name, config->librarySearchPaths,
                                   {".tbd", ".dylib", ".so"}))
         return path;
-      return findPathCombination("lib" + name, config->librarySearchPaths,
-                                 {".a"});
+      else if (std::optional<StringRef> path = findPathCombination(
+                   "lib" + name, config->librarySearchPaths, {".a"}))
+        return path;
+      return findPathCombination(name, config->librarySearchPaths, {""});
```

Comment:
How come this needs to be two separate calls to `findPathCombination`, instead of keeping one call and making the argument `{".a", ""}`)? IIUC, the difference will be in a scenario where your search path has `/foo` and `/bar`, the files `/foo/crt1.o` and `/bar/crt1.o.a` exist, and you do `-lcrt1.o`. Your current logic will find `/bar/crt1.o.a`, whereas a single call would find `/foo/crt1.o`. What does ld64 do? If you do need the separate calls to match ld64's behavior then it's worth adding a comment to explain that.

Reply:
are you suggesting the code be
```c++
      return findPathCombination("lib" + name, config->librarySearchPaths,
                                 {".a", ""});
```
Won't that look for `libcrt1.o` instead of `crt1.o`?

Reply:
You're right; that was a think-o on my part :) That being said, the scenario in my first comment (except with `/bar/libcrt1.o.a`) still applies; it's an edge case, but if we're matching ld64 behavior, we might as well match it as closely as reasonable.

Reply:
a quick test with both `libcrt1.o.a` and `crt1.o` shows ld64 prefers crt1.o

Reply:
Hmm. In that case, we probably want to instead change `findPathCombination` to take a list of prefixes, so that we can match ld64's search behavior?

Reply:
removing the object and trying again gives an error, ld64 wont link with the .a file at all

Reply:
Huh, that's interesting. I wonder if they have some special-cased logic around these specific extensions, or if they just won't append the `.a` if there's already any other extension present...

Reply:
Looks like ld64 checks if the library name ends in `.o` and if so, assumes it's a Csu support file, and looks for a file in the library search path with that exact name, not searching for any `libcrt1.o.a`.

If a Csu support file is not found ld64 automatically changes the target to MacOS 10.8 which stopped requiring Csu support files, then continues without linking the object.

Apples new linker does the same, but it has no concept of Csu support files and just links with it like it were a normal file, exiting with an error if it's not found.